base/tuple.jl

```diff
@@ -31,6 +31,42 @@ function _setindex(v, i::Integer, first, tail...)
 end
 _setindex(v, i::Integer) = ()
 
+getindex(into::Tuple{}, index::Tuple{}) = ()
+getindex(into::Tuple, index::Tuple{}) = ()
+getindex(into::Tuple{}, index::Tuple) = ()
+getindex(into::Tuple, index::Tuple) = begin
+    next = getindex(tail(into), tail(index))
+    ifelse(first(index), (first(into), next...), next)
+end
+
+setindex(old::Tuple{}, new, switch::Tuple{}) = ()
+setindex(old::Tuple{}, new, switch::Tuple) = ()
+setindex(old::Tuple, new, switch::Tuple{}) = old
+setindex(old::Tuple, new, switch::Tuple) =  begin
```

Comment:
Please use long-form function definitions for functions that span multiple lines. It's also clearer to surround returned tuples in parentheses when one of the arguments has to be splatted.

In this case, the function doesn't actually need to span multiple lines, as the `first_switch` definition is unnecessary. In keeping with the Base style, this should be
```julia
setindex(old::Tuple, new, switch::Tuple) =
    (ifelse(first(switch), new, first(old)), setindex(tail(old), new, tail(switch))...)
```